openquake/hazardlib/gsim/abrahamson_2014.py

```diff
@@ -0,0 +1,520 @@
+
+    #: Supported intensity measure types are spectral acceleration, peak
+    #: ground velocity and peak ground acceleration, see tables 4
+    #: pages 1036
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGA,
+        PGV,
+        SA
+    ])
+
+    #: Supported intensity measure component is orientation-independent
+    #: average horizontal :attr:`~openquake.hazardlib.const.IMC.GMRotI50`,
+    #: see page 1025.
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.RotD50
+
+    #: Supported standard deviation types are inter-event, intra-event
+    #: and total, see paragraph "Equations for standard deviations", page
+    #: 1046.
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL,
+        const.StdDev.INTER_EVENT,
+        const.StdDev.INTRA_EVENT
+    ])
+
+    #: Required site parameters are Vs30 and Z1.0, see table 2, page 1031
+    REQUIRES_SITES_PARAMETERS = set(('vs30', 'z1pt0', 'vs30measured'))
+
+    #: Required rupture parameters are magnitude, rake, dip, ztor, and width
+    #: (see table 2, page 1031)
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag', 'rake', 'dip', 'ztor', 'width'))
+
+    #: Required distance measures are Rrup, Rjb, Ry0 and Rx (see Table 2,
+    #: page 1031).
+    REQUIRES_DISTANCES = set(('rrup', 'rjb', 'rx', 'ry0'))
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        # extract dictionaries of coefficients specific to required
+        # intensity measure type and for PGA
+        C = self.COEFFS[imt]
+
+        # compute median sa on rock (vs30=1180m/s), used for site response
+        # term calculation
+        sa1180 = np.exp(self._get_sa_at_1180(imt, sites, rup, dists))
+
+        # get the mean value
+        mean = (self._get_basic_term(C, rup, dists) +
+                self._get_faulting_style_term(C, rup) +
+                self._get_site_response_term(C, imt, sites.vs30, sa1180) +
+                self._get_hanging_wall_term(C, dists, rup) +
+                self._get_top_of_rupture_depth_term(C, rup) +
+                self._get_soil_depth_term(C, sites.z1pt0, sites.vs30)
+                )
+        mean += self._get_regional_term(C, sites.vs30, imt, dists.rrup)
+
+        # get standard deviations
+        stddevs = self._get_stddevs(C, rup, sites, stddev_types, sa1180, dists)
+
+        return mean, stddevs
+
+    def _get_sa_at_1180(self, imt, sites, rup, dists):
+        """
+        Compute and return mean imt value for rock conditions
+        (vs30 = 1100 m/s)
+        """
+        # reference vs30 = 1180 m/s
+        vs30_1180 = np.ones_like(sites.vs30) * 1180
+        # reference shaking intensity = 0
+        ref_iml = np.zeros_like(sites.vs30)
+        C = self.COEFFS[imt]
```

Comment:
Do you need to call this twice? Why not just pass the initial C as an input to the function